src/utils/SaveDataUtils.ts

```diff
@@ -11,7 +22,24 @@ async function compress(dataString: string): Promise<Uint8Array> {
 
 async function decompress(binaryData: Uint8Array): Promise<string> {
   const decompressedReadableStream = new Blob([binaryData]).stream().pipeThrough(new DecompressionStream("gzip"));
-  return await new Response(decompressedReadableStream).text();
+  const reader = decompressedReadableStream.pipeThrough(new TextDecoderStream()).getReader();
+  let result = "";
+  // Use "done" here to stop Lint from showing error
+  const done = false;
+  try {
+    while (!done) {
+      const readResult = await reader.read();
+      if (readResult.done) {
+        break;
+      }
+      result += readResult.value;
+    }
```

Comment:
I've done this before, and I think there's a more elegant way of structuring this.

```suggestion
  const reader =  decompressedReadableStream.pipeThrough(new TextDecoderStream("utf-8", {fatal: true})).getReader();
  let result = "";
  try {
    for (let {value, done} = await reader.read(); !done; {value, done} = await reader.read()) {
        result += value;
    }
```

Also, there's no need for the `finally` since the reader immediately goes out of scope anyway (and in general, it is single-use.)